include/ignition/common/Battery.hh

```diff
@@ -59,13 +57,22 @@ namespace ignition
       /// \param[in] _battery Battery to copy.
       public: Battery(const Battery &_battery);
 
-      /// \brief Assignment operator
+      /// \brief Copy assignment operator
       /// \param[in] _battery The new battery
       /// \return a reference to this instance
       public: Battery &operator=(const Battery &_battery);
 
+      /// \brief Move constructor
+      /// \param[in] _battery Battery to move.
+      public: Battery(Battery &&_battery);
+
+      /// \brief Move assignment operator
+      /// \param[in] _battery The new battery
+      /// \return a reference to this instance
+      public: Battery &operator=(Battery &&_battery);
```

Comment:
Shouldn't the `ImplPtr` let us use rule-of-zero?

Reply:
In general, it will.  Unfortunately, `BatteryPrivate` has a `std::mutex` in it, which has no copy or move constructors, so we have to implement the full rule of 5.

In retrospect, I think that the locking should have happened outside of `Battery` by the downstream user, but I think that's a more substantial change that should come in a different PR.